examples/opensora_pku/opensora/models/diffusion/opensora/net_with_loss.py

```diff
@@ -204,43 +176,74 @@ def compute_loss(self, x, text_embed, encoder_attention_mask, attention_mask=Non
-        else:
-            temp_attention_mask, loss_mask = None, None
 
-        t = ops.randint(0, self.diffusion.num_timesteps, (x.shape[0],), dtype=ms.int32)
+        t = ops.randint(0, self.num_train_timesteps, (x.shape[0],), dtype=ms.int32)
         if get_sequence_parallel_state():
-            t = self.reduce_t(t) % self.diffusion.num_timesteps
-        x_t = self.diffusion.q_sample(x, t, noise=noise)
+            t = self.reduce_t(t) % self.num_train_timesteps
+        x_t = self.noise_scheduler.add_noise(x, noise, t)
 
         # latte forward input match
         # text embed: (b n_tokens  d) -> (b  1 n_tokens d)
         # text_embed = ops.expand_dims(text_embed, axis=1)
-        model_output = self.apply_model(
+        model_pred = self.apply_model(
             x_t,
             t,
             encoder_hidden_states=text_embed,
             attention_mask=attention_mask,
             encoder_attention_mask=encoder_attention_mask,
-            temp_attention_mask=temp_attention_mask,
             use_image_num=use_image_num,
         )
 
-        if loss_mask is not None:
-            model_output *= loss_mask
-
+        if self.prediction_type == "epsilon":
+            target = noise
+        elif self.prediction_type == "v_prediction":
+            target = self.noise_scheduler.get_velocity(x, noise, t)
+        elif self.prediction_type == "sample":
+            # We set the target to latents here, but the model_pred will return the noise sample prediction.
+            target = x
+            # We will have to subtract the noise residual from the prediction to get the target sample.
+            model_pred = model_pred - noise
+        else:
+            raise ValueError(f"Unknown prediction type {self.prediction_type}")
+        # comment it to avoid graph syntax error
+        # if attention_mask is not None and (attention_mask.bool()).all():
+        #     attention_mask = None
+        if get_sequence_parallel_state():
+            assert (attention_mask.bool()).all()
+            # assert attention_mask is None
+            attention_mask = None
         # (b c t h w),
-        B, C, F = x_t.shape[:3]
-        assert (
-            model_output.shape == (B, C * 2, F) + x_t.shape[3:]
-        ), f"model_output shape {model_output.shape} and x_t shape {x_t.shape} mismatch!"
-        model_output, model_var_values = ops.split(model_output, C, axis=1)
-
-        # Learn the variance using the variational bound, but don't let it affect our mean prediction.
-        vb = self._cal_vb(ops.stop_gradient(model_output), model_var_values, x, x_t, t)
-
-        loss = mean_flat((noise - model_output) ** 2) + vb
-        loss = loss.mean()
+        bsz, c, _, _, _ = model_pred.shape
+        if attention_mask is not None:
+            attention_mask = attention_mask.unsqueeze(1).float().repeat(c, axis=1)  # b t h w -> b c t h w
```

Comment:
why float?

Reply:
Torch code converts attention mask to float tensor here. I think it is to keep the multiplication of loss(fp32) and attention_mask (fp32) at the same data type: 
```python
loss = (loss * attention_mask).sum() / attention_mask.sum() 
```